src/protocol/dodo/state.rs

```diff
@@ -173,7 +170,11 @@ where
             .pool_abi
             .decode_output::<U256, _>("querySellBaseToken", simulation_result.result)
             .expect("DODO: Failed decoding get_amount_out result!");
-        Ok(GetAmountOutResult { amount: amount_out, gas: U256::from(simulation_result.gas_used) })
+        Ok(GetAmountOutResult {
+            amount: amount_out,
+            gas: U256::from(simulation_result.gas_used),
+            new_state: self.clone_box(), // WARNING: This is not correct but dodo is not being used
```

Comment:
Let's remove Dodo then

Reply:
Better than having a non-working dependency on a new package

Reply:
sure? I wanted to remove it some time ago but @kayibal was against it 😅